src/modules/map/components/cesium/CesiumMap.vue

```diff
@@ -11,25 +28,54 @@ import {
     Color,
     Cartesian3,
 } from 'cesium'
-import { addSwisstopoWMTSLayer } from './utils/imageryLayerUtils'
 import { mapGetters, mapState } from 'vuex'
 import { TERRAIN_URL } from './constants'
-import { IS_TESTING_WITH_CYPRESS } from '@/config'
+import { IS_TESTING_WITH_CYPRESS, WMTS_BASE_URL } from '@/config'
+import CesiumInternalLayer from './CesiumInternalLayer.vue'
+import GeoAdminWMTSLayer from '@/api/layers/GeoAdminWMTSLayer.class'
+import LayerTimeConfig from '@/api/layers/LayerTimeConfig.class'
+import { CURRENT_YEAR_WMTS_TIMESTAMP } from '@/api/layers/LayerTimeConfigEntry.class'
+
 export default {
+    components: { CesiumInternalLayer },
     provide() {
         return {
             // sharing cesium viewer object with children components
             getViewer: () => this.viewer,
         }
     },
+    data() {
+        return {
+            viewerCreated: false,
```

Comment:
is this really needed? As you create the Cesium instance in `beforeCreate()` lifecycle hook, it should always be set whenever the component is mounted, meaning the `v-if` above in the template should always be `true`

Reply:
yes, it is needed. We can't create a Cesium viewer in `beforeCreate` because Cesium requires a container for the viewer so Cesium instants are created when the component mounted